spark/interpreter/src/main/java/org/apache/zeppelin/spark/PySparkInterpreter.java

```diff
@@ -171,11 +171,12 @@ public void setInterpreterContextInPython() {
   // spark.pyspark.driver.python > spark.pyspark.python > PYSPARK_DRIVER_PYTHON > PYSPARK_PYTHON
   @Override
   protected String getPythonExec() {
-    if (!StringUtils.isBlank(getProperty("spark.pyspark.driver.python", ""))) {
-      return properties.getProperty("spark.pyspark.driver.python");
+    SparkConf sparkConf = getSparkConf();
+    if (StringUtils.isNotBlank(sparkConf.get("spark.pyspark.driver.python", ""))) {
+      return sparkConf.get("spark.pyspark.driver.python");
     }
-    if (!StringUtils.isBlank(getProperty("spark.pyspark.python", ""))) {
-      return properties.getProperty("spark.pyspark.python");
+    if (StringUtils.isNotBlank(sparkConf.get("spark.pyspark.python", ""))) {
+      return sparkConf.get("spark.pyspark.python");
     }
     if (System.getenv("PYSPARK_PYTHON") != null) {
       return System.getenv("PYSPARK_PYTHON");
```

Comment:
On line 175~179, "spark.pyspark.driver.python" takes precedence over "spark.pyspark.python".
But on line 181~185, "PYSPARK_PYTHON" takes precedence over "PYSPARK_DRIVER_PYTHON".

Is it intentional?

Reply:
Thanks for the careful review, this is a bug, fixed it now.